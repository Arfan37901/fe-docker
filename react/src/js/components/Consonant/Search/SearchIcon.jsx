import React from 'react';
import PropTypes from 'prop-types';

const SearchIcon = (props) => {
    const { onClick } = props;

    return (
        <button
            type="button"
            className="search-ico"
            onClick={onClick}
            tabIndex="0">
            <span />
            Click to search
        </button>
    );
};

export default SearchIcon;

SearchIcon.propTypes = {
    onClick: PropTypes.func.isRequired,
};
