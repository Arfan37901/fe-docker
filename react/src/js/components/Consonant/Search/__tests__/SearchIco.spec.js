import { screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import SearchIcon from '../SearchIcon';

import makeSetup from '../../Helpers/Testing/Utils/Settings';

const setup = makeSetup(SearchIcon, {
    onClick: jest.fn(),
});

describe('Consonant/SearchIco', () => {
    describe('Check snapshots', () => {
        test('should renders correctly', () => {
            const { tree } = setup();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('Interaction with UI', () => {
        test('should call onClick', () => {
            const { props: { onClick } } = setup();

            const iconElement = screen.getByText('Click to search');

            fireEvent.click(iconElement);

            expect(onClick).toBeCalled();
        });
    });
});
