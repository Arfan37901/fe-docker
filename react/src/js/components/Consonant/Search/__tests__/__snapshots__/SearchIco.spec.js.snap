// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/SearchIco Check snapshots should renders correctly 1`] = `
<button
  className="search-ico"
  onClick={[MockFunction]}
  tabIndex="0"
  type="button"
>
  <span />
  Click to search
</button>
`;
